### PR TITLE
MarkerStyle is considered immutable

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

as discussed in yesterdays dev call.

See also https://github.com/matplotlib/matplotlib/pull/17850#issuecomment-763195168.

### CODE REVIEW

File: lib/matplotlib/markers.py [hunk 1]
```diff
@@ -147,6 +147,9 @@ class MarkerStyle:
     """
     A class representing marker types.
 
+    Instances are considered immutable. If you need to change anything, create
```
Comment (anntzer) [inline lib/matplotlib/markers.py:150]: "Instances are immutable."?

File: lib/matplotlib/tests/test_marker.py [hunk 2]
```diff
@@ -14,26 +14,19 @@ def test_marker_fillstyle():
...
+        markers.MarkerStyle(np.array([[-0.5, 0, 1, 2, 3]]))
 
 
 def test_marker_path():
-    marker_style = markers.MarkerStyle()
-    path = Path([[0, 0], [1, 0]], [Path.MOVETO, Path.LINETO])
     # Checking this doesn't fail.
-    marker_style.set_marker(path)
+    path = Path([[0, 0], [1, 0]], [Path.MOVETO, Path.LINETO])
+    markers.MarkerStyle(path)
```
Comment (anntzer) [inline lib/matplotlib/tests/test_marker.py:29]: I'd consider (optionally) merging the three tests while you're at it.

File: lib/matplotlib/lines.py [hunk 3]
```diff
@@ -534,7 +534,7 @@ def set_fillstyle(self, fs):
 
             For examples see :ref:`marker_fill_styles`.
         """
-        self._marker.set_fillstyle(fs)
+        self.set_marker(MarkerStyle(self._marker.get_marker(), fs))
```
Comment (jklymak) [inline lib/matplotlib/lines.py:537]: I guess we never test `set_fillstyle`?  Can we use this as the opportunity to do so?

**Review by jklymak — COMMENTED** [review]
Overall makes sense to me, but it seems `lines.set_fillstyle` is not tested?  Or did you want to deprecate that as well?

File: lib/matplotlib/lines.py (hunk 3 above)
Comment (timhoffm) [reply lib/matplotlib/lines.py:537]: I'd like to have this deprecation be part of 3.4 and I don't know if I have the bandwidth to add tests soon. Therefore, I'd rather do this in a later PR.

**Comment by timhoffm** [comment]
As opposed to what I said in the dev call, we cannot easily deprecate `Lines.set/get_fillstyle`. I thought `get_marker()` would return the `MarkerStyle` so that we could do `line.get_marker().get_fillstyle()`. However, the used `MarkerStyle` object is not public. `get_marker()` does not return it but only the marker symbol, e.g. `'o'`. Therefore, I'd leave marker and fillstyle as "simple" Matplotlib properties of `Line2D` for now.

**Review by anntzer — APPROVED** [review]

**Review by jklymak — APPROVED** [review]

**Pull request merged**